content/docs/user-guide/dvc-files.md

```diff
@@ -237,14 +242,224 @@ stages:
 ```
 
 Stage commands are listed again in `dvc.lock`, in order to know when their
-definitions change in the `dvc.yaml` file.
+definitions change in `dvc.yaml`.
 
 Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
 (including [metrics](/doc/command-reference/metrics) and
 [plots](/doc/command-reference/plots) files) are also listed (per stage) in
-`dvc.lock`, but with an additional field to store the hash value of each file or
-directory tracked by DVC. Specifically: `md5`, `etag`, or `checksum` (same as in
-`deps` and `outs` entries of `.dvc` files).
+`dvc.lock`, but with an additional field with a hash of their last known
+contents. Specifically: `md5`, `etag`, or `checksum` are used (same as in `deps`
+and `outs` entries of `.dvc` files).
+
+Full <abbr>parameters</abbr> (key and value) are listed too (under `params`),
+grouped by parameters file. And in the case of templated/parameterized
+`dvc.yaml` files (see next section), their actual values are substituted into
+the `dvc.lock` YAML structure.
+
+### Parameterize `dvc.yaml`
+
+`dvc.yaml` supports a special templating format that enables reusing simple
+values from different sources in the YAML structure itself.
+
+Global parameters (file scope) can be defined either in a
+[params file](/doc/command-reference/params) (`params.yaml` by default), or in a
+top `vars` section of `dvc.yaml`. Both can coexist as long as `vars` don't
+overwrite any external param.
+
+Let's say we have a `params.yaml` file with the following contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+To use those values in `dvc.yaml`, you can use the `${}` expression anywhere in
+the file:
+
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --tresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
+```
+
+Alternatively, the same or other values can be included as `vars` like this:
+
+```yaml
+vars:
+  - models:
+    us:
+      thresh: 10
+  - desc: 'Reusable description'
+
+stages:
+  build-us:
+    desc: ${desc}
+    cmd: python train.py --tresh ${models.us.threshold} ...
+```
+
+The global `vars` section also accepts any number of parameters file names as
+entries as a way to include their contents for substituting in `dvc.yaml`, e.g.:
+
+```yaml
+vars:
+  - params.json
+  - myconfig.yaml
+  - myvar: value
+```
+
+> ⚠️ Note that the default `params.yaml` is always included (completely) first.
```

Comment:
get rid of compeletely or simplify - `all values ...`

Reply:
This has already been removed 👍